### PR TITLE
Amls 5096 - New Auth for Other 

### PR DESCRIPTION
**Pull request by slavvco**
Migration of other controllers (like Landing, Status and so on) to new auth. Includes updates to some services and auth action.

## Related / Dependant PRs?

- none

## How Has This Been Tested?

- unit tests, regression

## Checklist

- [x] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [ ] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/connectors/AmlsConnector.scala [hunk 1]
```diff
@@ -240,6 +207,18 @@ class AmlsConnector @Inject()(val httpPost: WSHttp,
...
+
+    Logger.debug(s"[AmlsConnector][savePayment]: Request to $postUrl with paymentId $paymentId")
+
+    httpPost.POSTString[HttpResponse](postUrl, paymentId)
+  }
+
+  @deprecated("to be removed after new auth changes implemented")
   def getPaymentByPaymentReference(paymentReference: String)
                                   (implicit hc: HeaderCarrier, ec: ExecutionContext, ac: AuthContext): Future[Option[Payment]] = {
     //TODO - deprecated by AuthAction.accountTypeAndId after new auth changes
```
Comment (matthewharris1) [inline app/connectors/AmlsConnector.scala]: Is there a need for this and @deprecated?

File: app/controllers/BacsConfirmationController.scala [hunk 2]
```diff
@@ -18,34 +18,33 @@ package controllers
 
 import cats.data.OptionT
 import cats.implicits._
+import config.AppConfig
 import connectors.{AmlsConnector, DataCacheConnector, _}
 import javax.inject.{Inject, Singleton}
 import models.businessdetails.{BusinessDetails, PreviouslyRegisteredYes}
 import services.{AuthEnrolmentsService, StatusService}
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
```
Comment (matthewharris1) [inline app/controllers/BacsConfirmationController.scala]: Not sure this import is needed

File: app/controllers/ConfirmationController.scala [hunk 3]
```diff
@@ -30,49 +31,42 @@ import services.{AuthEnrolmentsService, FeeResponseService, StatusService, _}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.frontend.auth.AuthContext
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
```
Comment (matthewharris1) [inline app/controllers/ConfirmationController.scala]: ^

File: app/controllers/LoginEventController.scala [hunk 4]
```diff
@@ -18,14 +18,15 @@ package controllers
 
 import javax.inject.{Inject, Singleton}
 import play.api.mvc.{Action, AnyContent}
-import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
+import utils.AuthAction
 import views.html.login_event
+
```
Comment (matthewharris1) [inline app/controllers/LoginEventController.scala]: any need for whitespace?

File: app/controllers/NotificationController.scala [hunk 5]
```diff
@@ -31,46 +31,45 @@ import services.{AuthEnrolmentsService, NotificationService, StatusService}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.frontend.auth.AuthContext
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
```
Comment (matthewharris1) [inline app/controllers/NotificationController.scala]: Need this import?

File: app/controllers/StatusController.scala [hunk 6]
```diff
@@ -31,66 +31,64 @@ import services._
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.frontend.auth.AuthContext
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
```
Comment (matthewharris1) [inline app/controllers/StatusController.scala]: import

File: app/services/LandingService.scala [hunk 7]
```diff
@@ -116,62 +112,63 @@ class LandingService @Inject() (
    * shouldn't be a problem as this should only happen when someone
```
Comment (matthewharris1) [inline app/services/LandingService.scala]: Need this comment?

File: app/services/SubmissionService.scala [hunk 8]
```diff
@@ -126,69 +119,80 @@ class SubmissionService @Inject()
...
   }
 
-  def update
-  (implicit
-   ec: ExecutionContext,
-   hc: HeaderCarrier,
-   ac: AuthContext
-  ): Future[AmendVariationRenewalResponse] = {
+  private def createSubscriptionRequestNewAuth(cache: CacheMap)
+                                       (implicit hc: HeaderCarrier, ec: ExecutionContext): SubscriptionRequest = {
```
Comment (matthewharris1) [inline app/services/SubmissionService.scala]: align params please

File: app/services/SubmissionService.scala [hunk 9]
```diff
@@ -126,69 +119,80 @@ class SubmissionService @Inject()
...
+      _ <- saveResponseNewAuth(credId, response, AmendVariationRenewalResponse.key, isRenewalAmendment = true)
     } yield response
   }
 
-  private def saveResponse[T](response: T, key: String, isRenewalAmendment: Boolean = false)
-                             (implicit ac: AuthContext, hc: HeaderCarrier, ex: ExecutionContext, fmt: Format[T]) = for {
-    _ <- cacheConnector.save[T](key, response)
-    c <- cacheConnector.save[SubmissionRequestStatus](SubmissionRequestStatus.key, SubmissionRequestStatus(true, Some(isRenewalAmendment)))
+  private def saveResponseNewAuth[T](credId: String, response: T, key: String, isRenewalAmendment: Boolean = false)
+                             (implicit hc: HeaderCarrier, ex: ExecutionContext, fmt: Format[T]) = for {
```
Comment (matthewharris1) [inline app/services/SubmissionService.scala]: ^

File: app/controllers/LoginEventController.scala (hunk 4 above)
Comment (slavvco) [reply app/controllers/LoginEventController.scala]: it was added automatically by IDE, but I will remove

File: app/services/LandingService.scala (hunk 7 above)
Comment (slavvco) [reply app/services/LandingService.scala]: not mine, leaving because it may need some attention

File: app/services/SubmissionService.scala (hunk 9 above)
Comment (slavvco) [reply app/services/SubmissionService.scala]: done

File: app/services/SubmissionService.scala (hunk 8 above)
Comment (slavvco) [reply app/services/SubmissionService.scala]: done

File: app/controllers/StatusController.scala (hunk 6 above)
Comment (slavvco) [reply app/controllers/StatusController.scala]: done

File: app/controllers/NotificationController.scala (hunk 5 above)
Comment (slavvco) [reply app/controllers/NotificationController.scala]: nope, done

File: app/controllers/LoginEventController.scala (hunk 4 above)
Comment (slavvco) [reply app/controllers/LoginEventController.scala]: done

File: app/controllers/ConfirmationController.scala (hunk 3 above)
Comment (slavvco) [reply app/controllers/ConfirmationController.scala]: done

File: app/controllers/BacsConfirmationController.scala (hunk 2 above)
Comment (slavvco) [reply app/controllers/BacsConfirmationController.scala]: done

File: app/connectors/AmlsConnector.scala (hunk 1 above)
Comment (slavvco) [reply app/connectors/AmlsConnector.scala]: removed

**Comment by slavvco** [comment]
approved so merging for testing in environments

**Pull request merged**